app/src/app/utils/helpers.ts

```diff
@@ -498,15 +498,35 @@ const filterFeatures = (
   filterLocked: boolean = true,
   additionalFilters: Array<(f: MapGeoJSONFeature) => boolean> = []
 ) => {
-  const {captiveIds, lockedFeatures, mapDocument, checkParentsToHeal, shatterIds} =
-    useMapStore.getState();
+  const {
+    activeTool,
+    captiveIds,
+    lockedFeatures,
+    mapDocument,
+    mapOptions,
+    checkParentsToHeal,
+    selectedZone,
+    shatterIds,
+    zoneAssignments,
+  } = useMapStore.getState();
   const parentIdsToHeal: MapStore['parentsToHeal'] = [];
   const filterFunctions: Array<(f: MapGeoJSONFeature) => boolean> = [...additionalFilters];
   if (captiveIds.size) {
     filterFunctions.push(f => captiveIds.has(f.id?.toString() || ''));
   }
-  if (filterLocked && lockedFeatures.size) {
-    filterFunctions.push(f => !lockedFeatures.has(f.id?.toString() || ''));
+  if (filterLocked) {
+    if (activeTool === 'brush' && mapOptions.lockPaintedAreas.includes(selectedZone)) {
+      return [];
+    } else if (mapOptions.lockPaintedAreas.length) {
+      const lockedAreas = mapOptions.lockPaintedAreas;
+      filterFunctions.push(
+        f =>
+          !lockedFeatures.has(f.id?.toString() || '') &&
+          !lockedAreas.includes(zoneAssignments.get(f.id?.toString() || '') || null)
+      );
+    } else if (lockedFeatures.size) {
```

Comment:
maybe it's too long since I looked at this - I thought that lockedFeatures was a set of IDs that have a locked color (from mapOptions.lockPaintedAreas), is it connected to shattering or county paint?